### PR TITLE
docs: add guide on retrieving user roles

### PR DESCRIPTION
**Pull request by dakshitha**
### Definition of Ready

- [ ] I am happy with the code
- [ ] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] No debug or dead code
- [ ] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

File: docs/docs/guides/integrate/retrieve-user-roles.md [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+## Assign roles and memberships
+
+Human or service users can be assigned roles. You can do this via the ZITADEL Console or the ZITADEL APIs. As mentioned earlier, there are two types of roles in ZITADEL. You can have your own application-specific roles, and alternatively, ZITADEL also has manager roles, such as `ORG_OWNER` and `IAM_OWNER`. 
+
+Follow the links below to assign roles to your users. 
+
+- [Add application roles via the ZITADEL Console](https://zitadel.com/docs/guides/manage/console/roles)
+- [Add manager roles via the ZITADEL Console](https://zitadel.com/docs/guides/manage/console/managers)
+- [Add application  roles via the ZITADEL Management API](https://zitadel.com/docs/category/apis/resources/mgmt/project-roles)
+- [Add manager roles to users via the ZITADEL Management API](https://zitadel.com/docs/category/apis/resources/mgmt/members)
```
Comment (hifabienne) [inline docs/docs/guides/integrate/retrieve-user-roles.md:21]:
```suggestion
- [Add application roles via the ZITADEL Console](/docs/guides/manage/console/roles)
- [Add manager roles via the ZITADEL Console](/docs/guides/manage/console/managers)
- [Add application roles via the ZITADEL Management API](/docs/category/apis/resources/mgmt/project-roles)
- [Add manager roles to users via the ZITADEL Management API](/category/apis/resources/mgmt/members)
```

File: docs/docs/guides/integrate/retrieve-user-roles.md [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+Roles can be requested via our auth and management APIs, from userinfo endpoint or ID token. Currently, manager roles cannot be directly included in the token. You will need to use the ZITADEL APIs to retrieve them.
+
+### Generate a token
+
+You must first of all generate a token for the user. If it’s a human user, he would be using a front-end application and logging in via the browser or device. An access token will be returned after they log in successfully. A machine user will use a script or other program to generate a token using the JWT profile or client credentials grant types. 
+
+How to generate a token: 
+
+- [Generate tokens for human users](https://zitadel.com/docs/guides/integrate/login-users)
+- [Generate tokens for service users](https://zitadel.com/docs/guides/integrate/serviceusers)
```
Comment (hifabienne) [inline docs/docs/guides/integrate/retrieve-user-roles.md:34]:
```suggestion
- [Generate tokens for human users](/docs/guides/integrate/login-users)
- [Generate tokens for service users](/docs/guides/integrate/serviceusers)
```

File: docs/docs/guides/integrate/retrieve-user-roles.md [hunk 3]
```diff
@@ -0,0 +1,470 @@
...
+Send the access token of the user as `Bearer Token` in the `Authorization` header:
+
+**cURL Request:**
+```bash
+curl --request GET \
+ --url $ZITADEL_DOMAIN/oidc/v1/userinfo
+ --header 'Authorization: Bearer <TOKEN>'
+```
+
+If the access token is valid, the information about the user (depending on the granted scopes) is returned. Check the [Claims page](https://zitadel.com/docs/apis/openidoauth/claims) for more details.
```
Comment (hifabienne) [inline docs/docs/guides/integrate/retrieve-user-roles.md:97]:
```suggestion
If the access token is valid, the information about the user (depending on the granted scopes) is returned. Check the [Claims page](/docs/apis/openidoauth/claims) for more details. 
```

File: docs/docs/guides/integrate/retrieve-user-roles.md [hunk 4]
```diff
@@ -0,0 +1,470 @@
...
+
+This request can be tested out in the following way:
+1. Select the **‘Assert Roles on Authentication’** checkbox.
+2. Do not include the roles claims in the scope.
+3. When you run the command, you will see that the roles were returned.
+4. If you unselect the **‘Assert Roles on Authentication’** checkbox, you will not see the roles.
+
+
+**2. Scope used:** `openid email profile urn:zitadel:iam:org:project:id:223281986649719041:aud urn:iam:org:project:roles urn:zitadel:iam:org:projects:roles`
+
```
Comment (hifabienne) [inline docs/docs/guides/integrate/retrieve-user-roles.md:147]: I would add a note that they should use the roles from "urn:zitadel:iam:org:project:223281986649719041:roles" iinstead of "urn:zitadel:iam:org:project:roles":
we do have both for backwards compatibility, but the one with the id is the new one

File: docs/docs/guides/integrate/retrieve-user-roles.md [hunk 5]
```diff
@@ -0,0 +1,470 @@
...
+2. Include the role claims in the scope as given.
+3. When you run the command, you will see the roles in the response.
+4. If you remove the role claims in the scope and run the command, you will not receive the roles.
+
+### Retrieve roles using the auth API
+
+Now we will use the auth API to retrieve roles from a logged in user using the user’s token
+The base URL is: **https://$ZITADEL_DOMAIN/auth/v1**
+
+Let’s start with a user who has multiple roles in different organizations in a multi-tenanted set up. You can use the logged in user’s token or the machine user’s token to retrieve the authorizations using the [APIs listed under user authorizations/grants in the auth API](https://zitadel.com/docs/category/apis/resources/auth/user-authorizations-grants).
```
Comment (hifabienne) [inline docs/docs/guides/integrate/retrieve-user-roles.md:195]:
```suggestion
Let’s start with a user who has multiple roles in different organizations in a multi-tenanted set up. You can use the logged in user’s token or the machine user’s token to retrieve the authorizations using the [APIs listed under user authorizations/grants in the auth API](/docs/category/apis/resources/auth/user-authorizations-grants). 
```

File: docs/docs/guides/integrate/retrieve-user-roles.md [hunk 6]
```diff
@@ -0,0 +1,470 @@
...
+
+Now we will use the auth API to retrieve roles from a logged in user using the user’s token
+The base URL is: **https://$ZITADEL_DOMAIN/auth/v1**
+
+Let’s start with a user who has multiple roles in different organizations in a multi-tenanted set up. You can use the logged in user’s token or the machine user’s token to retrieve the authorizations using the [APIs listed under user authorizations/grants in the auth API](https://zitadel.com/docs/category/apis/resources/auth/user-authorizations-grants). 
+
+**Scope used:** `openid urn:zitadel:iam:org:project:id:zitadel:aud`
+
+
+#### **1. [List my project roles](https://zitadel.com/docs/apis/resources/auth/auth-service-list-my-project-permissions)**
```
Comment (hifabienne) [inline docs/docs/guides/integrate/retrieve-user-roles.md:200]:
```suggestion
#### **1. [List my project roles](/docs/apis/resources/auth/auth-service-list-my-project-permissions)**
```

File: docs/docs/guides/integrate/retrieve-user-roles.md [hunk 7]
```diff
@@ -0,0 +1,470 @@
...
+
+```bash
+{
+  "result": [
+    "cfo"
+  ]
+}
+```
+
+#### **2.[List my ZITADEL permissions](https://zitadel.com/docs/apis/resources/auth/auth-service-list-my-zitadel-permissions)​**
```
Comment (hifabienne) [inline docs/docs/guides/integrate/retrieve-user-roles.md:223]:
```suggestion
#### **2.[List my ZITADEL permissions](/docs/apis/resources/auth/auth-service-list-my-zitadel-permissions)​**
```

File: docs/docs/guides/integrate/retrieve-user-roles.md [hunk 8]
```diff
@@ -0,0 +1,470 @@
...
+    "project.grant.read",
+    "project.grant.member.read",
+    "project.grant.user.grant.read",
+    "project.read:self",
+    "project.create"
+  ]
+}
+```
+
+#### **[3. List my authorizations/grants​](https://zitadel.com/docs/apis/resources/auth/auth-service-list-my-user-grants)**
```
Comment (hifabienne) [inline docs/docs/guides/integrate/retrieve-user-roles.md:271]:
```suggestion
#### **[3. List my authorizations/grants​](/docs/apis/resources/auth/auth-service-list-my-user-grants)**
```

File: docs/docs/guides/integrate/retrieve-user-roles.md [hunk 9]
```diff
@@ -0,0 +1,470 @@
...
+  ]
+}
+```
+
+### Retrieve roles using the management API
+Now we will use the management API to retrieve user roles under an admin user. 
+
+The base URL is: **https://$ZITADEL_DOMAIN/management/v1**
+
+In [APIs listed under user grants in the management API](https://zitadel.com/docs/category/apis/resources/mgmt/user-grants), you will see that you can use the management API to retrieve and modify user grants. The two API paths that we are interested in to fetch user roles are given below.
```
Comment (hifabienne) [inline docs/docs/guides/integrate/retrieve-user-roles.md:379]:
```suggestion
In [APIs listed under user grants in the management API](/docs/category/apis/resources/mgmt/user-grants), you will see that you can use the management API to retrieve and modify user grants. The two API paths that we are interested in to fetch user roles are given below.
```

File: docs/docs/guides/integrate/retrieve-user-roles.md [hunk 10]
```diff
@@ -0,0 +1,470 @@
...
+
+**Scope used:** `openid urn:zitadel:iam:org:project:id:zitadel:aud`
+
+#### **1. [Search user grants](https://zitadel.com/docs/apis/resources/mgmt/management-service-list-user-grants)​**
```
Comment (hifabienne) [inline docs/docs/guides/integrate/retrieve-user-roles.md:383]:
```suggestion
#### **1. [Search user grants](/docs/apis/resources/mgmt/management-service-list-user-grants)​**
```

File: docs/docs/guides/integrate/retrieve-user-roles.md [hunk 11]
```diff
@@ -0,0 +1,470 @@
...
+      "projectId": "223281986649719041",
+      "projectName": "HR",
+      "preferredLoginName": "david.wallace",
+      "userType": "TYPE_HUMAN"
+    }
+  ]
+}
+```
+
+#### **2. [User grant by ID](https://zitadel.com/docs/apis/resources/mgmt/management-service-get-user-grant-by-id)​**
```
Comment (hifabienne) [inline docs/docs/guides/integrate/retrieve-user-roles.md:452]:
```suggestion
#### **2. [User grant by ID](/docs/apis/resources/mgmt/management-service-get-user-grant-by-id)​** 
```

File: docs/docs/guides/integrate/retrieve-user-roles.md (hunk 4 above)
Comment (dakshitha) [reply docs/docs/guides/integrate/retrieve-user-roles.md:150]: Done

**Review by hifabienne — APPROVED** [review]

**Pull request merged**